### PR TITLE
[ACS-5436] Logical search final version

### PR DESCRIPTION
**Pull request by MichalKinas**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

Logical filter has been redesigned to use 4 inputs without chips. https://alfresco.atlassian.net/browse/ACS-5436
search-logical-filter

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/content-services/components/search-logical-filter.component.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Last reviewed: 2023-06-01
 
 # [Search Logical Filter component](../../../lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts "Defined in search-logical-filter.component.ts")
 
-Implements a [search widget](../../../lib/content-services/src/lib/search/models/search-widget.interface.ts) consisting of 3 chip inputs representing logical conditions to form search query from.
+Implements a [search widget](../../../lib/content-services/src/lib/search/models/search-widget.interface.ts) consisting of 4  inputs representing logical conditions to form search query from.
```
Comment (AleksanderSklorz) [inline docs/content-services/components/search-logical-filter.component.md:10]: double space after 4

File: lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts [hunk 2]
```diff
@@ -39,26 +40,22 @@ export interface LogicalSearchCondition extends LogicalSearchConditionEnumValued
...
     id: string;
     settings?: SearchWidgetSettings;
     context?: SearchQueryBuilderService;
     startValue: string;
-    displayValue$: Subject<string> = new Subject<string>();
-    resetObservable = this.reset$.asObservable();
+    searchCondition: LogicalSearchCondition;
+    fields = Object.keys(LogicalSearchFields);
     LogicalSearchFields = LogicalSearchFields;
+    displayValue$: Subject<string> = new Subject<string>();
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts:50]: you don't need repeat type at the left side

**Review by DenysVuika — APPROVED** [review]

**Review by AleksanderSklorz — APPROVED** [review]

**Comment by AleksanderSklorz** [comment]
@MichalKinas please fix also code smells: 
image

File: docs/content-services/components/search-logical-filter.component.md (hunk 1 above)
Comment (MichalKinas) [reply docs/content-services/components/search-logical-filter.component.md]: Removed

File: lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts (hunk 2 above)
Comment (MichalKinas) [reply lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts]: Right, removed

**Comment by MichalKinas** [comment]
> @MichalKinas please fix also code smells: image

Done, thanks

**Pull request merged**